### DISCUSSION TITLE
Advice for using multiple TileLayers

### DISCUSSION THREAD

**Discussion by cornhundred** [visgl/deck.gl#5677] [Q&A] [answered]
Hi, I would like to ask for some advice on our TileLayer use case. We want to be able to switch between showing different TileLayers (e.g. six layers where each layer is a microscopy image taken at a different Z depth). I have tried two approaches but each has their downsides:

* Approach 1: Have a single TileLayer that has its getTileData URL modified (e.g. using updateTriggers) 
    * this approach works, but we get flashing between image updates
* Approach 2: Make multiple TileLayers with a `visible` prop that is toggled
    * this approach works but zooming appears to be a little slower and we also occasionally see some flashing, which seems to be caused by a layer having to get data once it is set to visible

I think we will go with approach 2, but we would like to know if there is a way to reduce flashing between updates we sometimes see. For instance could we set the caching to pre-download images for all layers (or at least a low res image at all layers)?

**Comment by ibgreen** [comment, marked as answer]
When toggling layers, did you try just setting the `visible` prop instead of removing / adding layers. That prevents the layers from being destroyed / recreated.

**Reply by cornhundred** [reply]
Hi @ibgreen, yes I tried toggling the visible prop. Based on my reading of the docs it seems this is the preferred approach, but we were still running into some flashing (e.g. white background) when a layer was set to visible. However, once each layer was toggled through (using a slider) the flashing was gone since it did not have to fetch the data again. I'm wondering if there is a way to have each layer pre-cache something (even if it is not visible) to reduce this flashing.

**Reply by ibgreen** [reply]
Yes, I think adding a way to make tile layers start loading even when not visible could make some sense. Other layers load their data as soon as they are created. 

If so they should probably keep loading new tiles as the viewport changes while not visible. The downside is that loading five invisible tile layer will reduce loading performance of the visible layer (due to network congestion and some parsing overhead) 

Some additional integration with the RequestScheduler could help there as it supports priorities on requests.

Depending on how comfortable you are with deck.gl code, you could try to fork or subclass the TileLayer.
